spidermon/contrib/scrapy/monitors/monitors.py

```diff
@@ -571,9 +579,13 @@ def _get_jobs(self, states, number_of_jobs):
                 count=count,
                 has_tag=tags or None,
             )
-            total_jobs.extend(current_jobs)
 
-            if len(current_jobs) < MAX_API_COUNT or len(total_jobs) == number_of_jobs:
+            for job in current_jobs:
```

Comment:
**thought:** This makes sense since it seems we can't filter by close_reason in `client.spider.jobs.list()`.

I wonder if it makes sense to retrieve `MAX_API_COUNT` count jobs each time from the API and append to `total_jobs` the jobs that meet the criteria until we reach `number_of_jobs`. This should result in fewer API calls, but those calls will be larger. What do you think, @VMRuiz?

Reply:
If the default close reason is `finished` I would keep it as it is. As the most common scenario will be that you fetch the last few jobs and those will have the finished status. Even in the case where a second call is needed - due to a few unexpected failed jobs in between - it could still be faster for lower volumes than making 1 very big call.   

On the contrary, if it's expected that uncommon close reasons like `timeout` are used, then it makes sense to fetch as many jobs as possible, as they will be few or even none.